source/argument_parser.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+
+from interface import (ArgumentParserInterface, InputReaderInterface, OutputFormatterInterface,
+                       TaskExecutorInterface, Arguments, LibraryComponents)
+
+from source import (DirectoryReader, DirectoryWatcher, OutputFormatterFasta,
+                    OutputFormatterFastq, SequentialTaskExecutor, ParallelTaskExecutor)
+
+
+class ArgumentParser(ArgumentParserInterface):
```

Comment:
I see no reason for this to be class if it's only interaction is create it & then call parse_arguments once. I would make it just a function `parse_arguments`

Reply:
Well, it also initializes LibraryComponents class. Instead of returning it all I figured, it would be more understandable, if I designed it as a class. Also, if someone else will want to use that library in a future and he will need to implement his own ArgumentParser, he can simply create a subclass and generally just use existing code infrastructure.

Reply:
I would suggest using against patterns like subclass. OOP, as done by many, is done wrong. In most cases, functional design fares way better in terms of maintainability, clarity and testability. Perhaps checou out some Sandi Metz presentations, they are usually good inspiration

Reply:
I agree that this design decision is just silly. But I have a question. I don't like about python libraries that internal code is just mixed with API methods and the only distinction between them is just an underscore at the beginning of a function name. I really liked here that all of the implementation is separated from the library interface and it's clear for a new-comer what the interface looks like and what is the intended usage without spending too much time on trying to understand what the f*** it does. Just look at that:
https://github.com/nanoporetech/ont_fast5_api
It's just tedious piece of code for me to use.

If I use a functional design for this component, is there any preferred pythonic way to declare but not implement functions a thus create pretty interface? Or should I just create a commented document describing the functionality in interface folder? Maybe it's not that necessary in this small component, but I would like to know for a future reference. Thanks!

Reply:
Can we also discuss this one @ppershing ?